db-service/lib/deep-queries.js

```diff
@@ -35,7 +36,42 @@ async function onDeep(req, next) {
     if (query.UPDATE) return this.onUPDATE({ query })
     if (query.DELETE) return this.onSIMPLE({ query })
   }))
-  return res[0] ?? 0 // TODO what todo with multiple result responses?
+
+  let summedResult = 1
+  // only INSERTS
+  if (res.length > 1 && res[0].results) {
+    const transitions = getTransition(req.query.target, this, false, req.query.cmd || 'DELETE')
+    for (const result of res) {
+      if (transitions.target === result?.query?.target) {
+        summedResult = result
+        break
+      }
+    }
+
+    if (typeof summedResult === 'object') {
+      // sum up the affected rows and entries
+      summedResult.affectedRows = res.slice(1).reduce((totalRows, result) => {
+        if (result.query?.target === transitions.target) {
+          summedResult.query.INSERT.entries.push(...result.query.INSERT.entries)
+          return totalRows + result.affectedRows
+        }
+        return totalRows
+      }, summedResult.affectedRows || 0)
+
+      // summ up the results
+      summedResult.results = res.slice(1).reduce(
+        (resultsArray, result) => {
+          if (result.results) {
+            resultsArray.push(result.results[0])
+          }
+          return resultsArray
+        },
+        summedResult.results.length ? summedResult.results : [summedResult.results],
+      )
+    }
+  }
+  
+  return summedResult ? summedResult : (res[0] ?? 0) // TODO what todo with multiple result responses?
```

Comment:
I have a feeling this could be done a bit easier.

```suggestion
  return beforeData.length || Array.isArray(req.data) ? req.data.length : 1
```

Reply:
i dont think so. The tests expects the result structured like this:
```json
{
"affectedRows": 1,
"results": [{
  "changes": 1,
  "lastInsertRowid": 8319
}],
"query": ...
}
```
Therefore we would have to sum up res.results and res.affectedRows